rsconf/component/__init__.py

```diff
@@ -80,6 +84,55 @@ def build_write(self):
         self._root_bash.extend(self._root_bash_aux)
         self.buildt.write_root_bash(self.name, self._root_bash)
 
+    def gen_host_and_identity_ssh_keys(
+        self, j2_ctx, visibility, password_filename=None
+    ):
+        @contextlib.contextmanager
```

Comment:
Too complicated and unnecessary work. You can write the passphrase before the generate happens. Logic should be:

```py
for f:
    if not f.exists():
        process_call(
        ...
       _pass() if encrypt_identity and f == res.identity_f else "",
```
_pass would be simpler:

```py    
def _pass():
     s = db.secret_path(j2_ctx, f"{self.module_name}_ssh_passphrase.json", visibility=visibility)
     o = pkjson.load_any(s) if s.exists() else PKDict()
     p = o.get(self.user_name)
     if p is None:
         o[self.user_name] = p = db.random_string()
         pkjson.dump_pretty(o, filename=s)
     return p
```

Reply:
Much better, thanks 5cf71b2